gpflow/models/sgpr.py

```diff
@@ -150,46 +151,95 @@ class SGPR(SGPRBase):
 
     """
 
+    CommonTensors = namedtuple("CommonTensors", ["A", "B", "LB", "AAT", "L"])
+
+
     def maximum_log_likelihood_objective(self) -> tf.Tensor:
         return self.elbo()
-
-    def elbo(self) -> tf.Tensor:
+    
+    def _common_calculation(self):
```

Comment:
https://github.com/GPflow/GPflow/pull/1706#discussion_r693400410

Reply:
Let me know if you think it is worth pushing for this change in this PR.

Reply:
I guess it is fine for now and should perhaps be a separate PR. I do think it would simplify the code in a few places, and perhaps make it easier to do something in the future like cache values so that prediction is less expensive (i.e. avoiding redundant cholesky decompositions). But that is beyond the scope of this PR.